### ISSUE TITLE
msg sent routine never executed

### ISSUE THREAD

**Issue by salvq** [TheCatLady/docker-qnap-pushover#4]
Hi, I just tried the docker however it was not working.

For some reason I am not getting notification to my mobile, or better say routine after `if hasIncluded and not hasExcluded:` is never executed

```
[/etc/logs] # docker run \
> --name=qnap-pushover \
> -e TZ=America/New_York \
> -e LOG_LEVEL=DEBUG \
> -e NOTIFY_LEVEL=INFO \
> -e POLL_INTERVAL=10 \
> -e INCLUDE= \
> -e EXCLUDE= \
> -e TESTING_MODE=true \
> -e PUSHOVER_TOKEN=YYY \
> -e PUSHOVER_RECIPIENT=XXX \
> -v qnap-pushover:/data \
> -v /etc/logs/event.log:/event.log:ro \
> --restart always \
> thecatlady/qnap-pushover

 _,         _,                     ,  |)    _        _  ,_
/ |  /|/|  / |  |/\_----|/\_|  |  / \_|/\  / \_|  |_|/ /  |
\/|_/ | |_/\/|_/|_/     |_/  \/|_/ \/ |  |/\_/  \/  |_/   |/
  |)           (|      (|
https://github.com/TheCatLady/docker-qnap-pushover

INFO: NOTIFY_LEVEL is set to INFO. All system log events will trigger notifications.
INFO: Poll interval is set to 10 seconds.
INFO: Only system log events containing the following keywords will trigger notifications:
INFO: System log events containing the following keywords will not trigger notifications:
WARNING: Testing mode is enabled. The last 10 system log events will be re-queued for processing on EVERY container start.
INFO: Using Pushover application API token YYY and recipient user/group key(s) XXX.
INFO: Read the last-processed event ID from the data file. (909)
INFO: Testing mode is enabled. Re-queuing last 10 system log events for processing.
INFO: 10 new system log events detected in event.log.
INFO: Successfully processed 10 system log events.
INFO: Wrote the last-processed event ID to the data file. (909)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (909)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (909)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (909)

^CTraceback (most recent call last):
  File "/main.py", line 247, in <module>
    time.sleep(POLL_INTERVAL)
KeyboardInterrupt
[/etc/logs] #

```

When I update your script with simple `pushover_client.send_message("Hello!", title="Hello")` in upper section of script I am being notified. 

Shame I did not have a time to troubleshoot more, do you know where is the problem that the routine of triggering the msg is skipped ? I have a plan to troubleshoot during weekend, if you do so sooner I will be glad ;-)

Thanks

**Comment by TheCatLady** [comment]
@salvq Hi there!  Can you try pulling the pre-built image, and removing the INCLUDE and EXCLUDE environment variables if you aren't filtering the events?

**Comment by TheCatLady** [comment]
@salvq Hi there!  I've pushed a bug fix to the repo.  Can you try pulling the pre-built image?  I.e.:

```
docker stop qnap-pushover
docker rm qnap-pushover
docker image prune
docker run -d \
  --name=qnap-pushover \
  -e TZ=America/New_York `#optional` \
  -e LOG_LEVEL=DEBUG \
  -e NOTIFY_LEVEL=INFO \
  -e TESTING_MODE=True \
  -e PUSHOVER_TOKEN=<Pushover application API token> \
  -e PUSHOVER_RECIPIENT=<Pushover user and/or group key(s)> \
  -v qnap-pushover:/data \
  -v /etc/logs/event.log:/event.log:ro \
  --restart always \
  thecatlady/qnap-pushover
```

Thanks for bringing this issue to my attention.  Please test the new image and let me know if you still have issues.

**Comment by TheCatLady** [comment]
@salvq Hi there!  I've pushed a bug fix to the repo.  Can you try pulling the pre-built image?  I.e.:

```
docker stop qnap-pushover
docker rm qnap-pushover
docker image prune
docker run -d \
  --name=qnap-pushover \
  -e TZ=America/New_York `#optional` \
  -e LOG_LEVEL=DEBUG \
  -e NOTIFY_LEVEL=INFO \
  -e TESTING_MODE=True \
  -e PUSHOVER_TOKEN=<Pushover application API token> \
  -e PUSHOVER_RECIPIENT=<Pushover user and/or group key(s)> \
  -v qnap-pushover:/data \
  -v /etc/logs/event.log:/event.log:ro \
  --restart always \
  thecatlady/qnap-pushover
```

Thanks for bringing this issue to my attention.  Please test the new image and let me know if you still have issues.

**Comment by salvq** [comment]
Hi, just tried to pull new image and I receive a test notification however docker has been stopped after few msg received

```
https://github.com/TheCatLady/docker-qnap-pushover

INFO: NOTIFY_LEVEL is set to INFO. All system log events will trigger notifications.
INFO: Poll interval is set to 10 seconds.
INFO: INCLUDE keyword filter is not set.
INFO: EXCLUDE keyword filter is not set.
WARNING: Testing mode is enabled. The last 10 system log events will be re-queued for processing on EVERY container start.
INFO: Using Pushover application API token YYY and recipient user/group key(s) YYY.
INFO: Read the last-processed event ID from the data file. (921)
INFO: Testing mode is enabled. Re-queuing last 10 system log events for processing.
INFO: 10 new system log events detected in event.log.
DEBUG: Starting new HTTPS connection (1): api.pushover.net:443
DEBUG: https://api.pushover.net:443 "POST /1/messages.json?message=%3Cfont+color%3D%22grey%22%3E%3Cb%3EMultimedia+Console+enabled%3C%2Fb%3E%3C%2Ffont%3E%3Csmall%3E%3C%2Fsmall%3E&user=XXX&html=1&priority=-1&timestamp=1610490471&title=App+Center&token=YYY HTTP/1.1" 200 None
CRITICAL: 'NameError' object has no attribute 'strerror'
[~] # 
```

When I run without test flag, also docker stopped when new msg in NAS was generated (msg has not been sent)

```
INFO: NOTIFY_LEVEL is set to INFO. All system log events will trigger notifications.
INFO: Poll interval is set to 10 seconds.
INFO: INCLUDE keyword filter is not set.
INFO: EXCLUDE keyword filter is not set.
INFO: Testing mode is not enabled.
INFO: Using Pushover application API token YYY and recipient user/group key(s) XXX.
INFO: Read the last-processed event ID from the data file. (921)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (921)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (921)
INFO: No new system log events detected in event.log.
INFO: Wrote the last-processed event ID to the data file. (921)
INFO: 1 new system log events detected in event.log.
CRITICAL: 'NameError' object has no attribute 'strerror'
```

**Comment by TheCatLady** [comment]
@salvq Can you pull the latest image and try again?  There was a bug in the exception handling, but I suspect there's an issue with your Pushover credentials.

**Comment by TheCatLady** [comment]
@salvq Can you pull the latest image and try again?  There was a bug in the exception handling, but I suspect there's an issue with your Pushover credentials.  With the bug fix, you should be able to see the error message in the log output.

**Comment by TheCatLady** [comment]
@salvq Can you pull the latest image and try again?  I committed a few bug fixes in https://github.com/TheCatLady/docker-qnap-pushover/commit/42641fc1c2b58602bc8793055ae9e9de49d39c76.

**Comment by salvq** [comment]
> @salvq Can you pull the latest image and try again? I committed a few bug fixes in [42641fc](https://github.com/TheCatLady/docker-qnap-pushover/commit/42641fc1c2b58602bc8793055ae9e9de49d39c76).

Hi, just tested and seems to working fine, getting messages on the mobile.

Just a quick one, what the includes / excludes really means, I am bit confused where it is being compared to in qnap log:

- example system event log does includes fields like application, category, content
- example system access log does includes fields like connection type or action

Thanks

![image](https://user-images.githubusercontent.com/43242348/104430452-fa678900-5586-11eb-86f2-56116f7b34c6.png)

**Comment by TheCatLady** [comment]
@salvq The filter applies to keywords in the event description only (the "Content" column).  But you'll notice that the event description almost always has the application name at the beginning, enclosed in brackets.

Closing this issue since it has been resolved.  Please open a new issue for any other questions/problems you have!  Thanks 😸
